src/app/app.component.ts

```diff
@@ -16,69 +16,54 @@
  */
 
 import { Component } from "@angular/core";
-import { NavigationEnd, Router } from "@angular/router";
-import { filter, take } from "rxjs/operators";
-import { ConfigService } from "./core/config/config.service";
+import { map, mergeMap, tap } from "rxjs/operators";
 import { LoginStateSubject } from "./core/session/session-type";
 import { LoginState } from "./core/session/session-states/login-state.enum";
+import { DemoDataInitializerService } from "./core/demo-data/demo-data-initializer.service";
+import { environment } from "environments/environment";
+import { SetupService } from "./core/setup/setup.service";
+import { from, merge, Observable, of } from "rxjs";
 
 /**
  * Component as the main entry point for the app.
  * Actual logic and UI structure is defined in other modules.
  */
 @Component({
   selector: "app-root",
-  template: `@if (
-      !configReady && (loginState | async) === LoginState.LOGGED_IN
-    ) {
-      <app-application-loading></app-application-loading>
-    } @else if (configFullscreen) {
-      <router-outlet></router-outlet>
-    } @else {
+  template: `
+    @if (configReady$ | async) {
       <app-ui></app-ui>
-    }`,
+    } @else {
+      <app-application-loading></app-application-loading>
+    }
+  `,
   // eslint-disable-next-line @angular-eslint/prefer-standalone
   standalone: false,
 })
 export class AppComponent {
-  configFullscreen: boolean = false;
-  configReady: boolean = false;
+  configReady$: Observable<boolean>;
 
   constructor(
-    private router: Router,
-    private configService: ConfigService,
-    protected loginState: LoginStateSubject,
+    private loginState: LoginStateSubject,
+    private demoDataInitializer: DemoDataInitializerService,
+    private setupService: SetupService,
   ) {
-    this.detectConfigReadyState();
+    this.configReady$ = this.loginState.pipe(
+      // if logged out, we don't wait for config and treat this separately
+      map((loginState) => loginState !== LoginState.LOGGED_IN),
+      // immediately switch the state based on loginState but then take time for config readiness
+      mergeMap((loggedOut) => {
```

Comment:
@Abhinegi2 I decided to move the full-screen mode of admin views to the UI component for better UX and simpler logic here. Do have a look whether my code makes sense